R/utils.R

```diff
@@ -252,3 +252,21 @@ run_safely <- function(..., fun) {
   return(result)
 }
 
+
+#' Ensure That an Object is a Data Table
+#' @description This function ensures that an object is a data table.
+#' If the object is not a data table, it is converted to one. If the object
+#' is a data table, a copy of the object is returned.
+#' @param data An object to ensure is a data table
+#' @return A data table
+#' @keywords internal
+#' @importFrom data.table copy is.data.table as.data.table
+ensure_data.table <- function(data) {
+  if (!is.data.table(data)) {
```

Comment:
linting issues from CI to fix.

Reply:
this function reproduces the pattern you have in several places in your code so you should go back and use it if you really need it (vs just using as.data.table which I believe always makes a copy?

Reply:
I think I use it everywhere. Reason to use this over `as.data.table()` is that it preserves other classes. But maybe that's not necessary everywhere.